internal/provider/data_source_integrations.go

```diff
@@ -0,0 +1,178 @@
+	ConnectorID  types.String              `tfsdk:"connector_id"`
+	Integrations []models.IntegrationModel `tfsdk:"integrations"`
+}
+
+func (d *integrationsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integrations"
+}
+
+func (d *integrationsDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Use this data source to get list integrations.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Integration type to filter by",
+				Optional:            true,
+			},
+			"connector_id": schema.StringAttribute{
+				MarkdownDescription: "Integration connector id to filter by",
+				Optional:            true,
+			},
+			"integrations": schema.ListNestedAttribute{
+				Computed:            true,
+				MarkdownDescription: "List of integrations",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: IntegrationDataSourceAttributes(),
+				},
+			},
+		},
+	}
+}
+
+func (d *integrationsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	d.provider, resp.Diagnostics = toProvider(req.ProviderData)
+}
+
+func (d *integrationsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var model integrationsDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "Read integrations")
+	response, _, err := d.provider.client.IntegrationsApi.ListIntegrationsV2(ctx).
+		Execute()
+	if err != nil {
+		if apiError, ok := err.(*apono.GenericOpenAPIError); ok {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to list integrations, error: %s, body: %s", apiError.Error(), string(apiError.Body())))
+		} else {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to list integrations: %s", err.Error()))
+		}
+
+		return
+	}
+
+	model.ID = types.StringValue(id.UniqueId())
+	for _, integration := range response.Data {
+		if !model.Type.IsNull() && integration.Type != model.Type.ValueString() {
+			continue
+		}
+
+		if !model.ConnectorID.IsNull() && *integration.ProvisionerId.Get() != model.ConnectorID.ValueString() {
+			continue
+		}
+
+		m, diagnostics := models.ConvertToIntegrationModel(ctx, &integration)
+		resp.Diagnostics.Append(diagnostics...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		model.Integrations = append(model.Integrations, *m)
+	}
+
+	// Save state into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+}
+
+func IntegrationDataSourceAttributes() map[string]schema.Attribute {
```

Comment:
can't we reuse the resource integrations?

Reply:
No, the schema here is of the data source package: `github.com/hashicorp/terraform-plugin-framework/datasource/schema` while in the resource is of the resource package: `github.com/hashicorp/terraform-plugin-framework/resource/schema`